python/lsst/daf/butler_migrate/butler_attributes.py

```diff
@@ -241,3 +242,33 @@ def replace_dimensions_json(self, universe_version: int) -> None:
         """
         dimensions = load_historical_dimension_universe_json(universe_version)
         self.update(_DIMENSIONS_JSON_KEY, dimensions)
+
+
+def _is_expected_dimensions_json_mismatch(expected_json: str, actual_json: str) -> bool:
+    # Return `True` if the two dimension universe JSON strings differ only in
+    # ways expected because of the history of this data.  Older repositories
+    # that have been previously migrated have some documentation strings that
+    # don't match the current dimension universe because of how dimension
+    # universes were patched in earlier migrations.
+
+    diff = compare_json_strings(expected_json, actual_json, diff_style="ndiff")
+    if diff is None:
+        return True
+
+    return all(_is_expected_diff_line(line) for line in diff.splitlines())
+
+
+def _is_expected_diff_line(line: str) -> bool:
+    # ndiff prefix for matching lines and "hint" lines.
+    if line.startswith("  ") or line.startswith("? "):
+        return True
+
+    # Lines containing only docstring changes.
+    if re.match(r'^[-+]\s+"doc":', line):
+        return True
+
+    # Empty line.
+    if line.strip() == "":
+        return True
+
+    return False
```

Comment:
Feels a bit fragile. Would it be better to just ignore doc strings when doing diff (maybe optionally)? You could load JSON from string into dict, drop all `doc` keys recursively and compare the result.